src/__experimental__/components/form/form.component.js

```diff
@@ -128,16 +134,28 @@ class FormWithoutValidations extends React.Component {
   }
 
   addUnsavedWarningListener = () => {
+    this.unsavedListener = true;
     this._window.addEventListener('beforeunload', this.checkIsFormDirty);
   }
 
   removeUnsavedWarningListener = () => {
+    this.unsavedListener = false;
     this._window.removeEventListener('beforeunload', this.checkIsFormDirty);
   }
 
+  /** Returns true if any input has value */
+  checkFormDataExists() {
+    const { formInputs } = this.state;
+    return Object.keys(formInputs).map(id => (!!formInputs[id].length)).includes(true);
```

Comment:
I think there's a potential risk of `Object.keys(formInputs)` throwing a `TypeError: Cannot convert undefined or null to object` if `this.state.formInputs` hasn't been initialised :warning:

I realise that currently there is a safety check of `this.state.formInputs` in `checkIsFormDirty()`:
```js
if (this.state.formInputs) {
  this.setState({ isDirty: this.checkFormDataExists() });
}
```
But to be completely safe, I think it'd be good to add a safety check here. Something like this:
```js
checkFormDataExists() {
  const { formInputs } = this.state;
  if (!formInputs) {
    return false;
  }
  …
```

Reply:
Yeah I was 50/50 about including the extra guard, happy to add it in

Reply:
Many thanks :white_check_mark: